shaq_overflow/app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,49 @@
+class QuestionsController < ApplicationController
+
+  def index
+    @questions = Question.all
+    if params[:order] == "votes"
+      @questions = Question.all.sort_by {|q| q.votes.count }.reverse
+    elsif params[:order] == "trending"
+      @questions = Question.all.sort_by {|q| [q.elapsed, -1 * q.votes.count] }
+    else
```

Comment:
This might become unmanageable if you have large numbers of questions. Think about doing this in ActiveRecord directly with a group_by / count / limit query.